src/main/java/com/microsoft/sqlserver/jdbc/SQLServerBulkCopy.java

```diff
@@ -1839,7 +1840,9 @@ private void validateColumnMappings() throws SQLServerException {
                 // if no mapping is provided for csv file and metadata is missing for some columns throw error
                 if (null != serverBulkData) {
                     Set<Integer> columnOrdinals = serverBulkData.getColumnOrdinals();
-                    Iterator<Integer> columnsIterator = columnOrdinals.iterator();
+                    List<Integer> sortedList = new ArrayList<>(columnOrdinals);
+                    Collections.sort(sortedList);
+                    Iterator<Integer> columnsIterator = sortedList.iterator();
```

Comment:
If the `columnMetadata` Map objects implemented in the SQLServerBulkRecord classes were `TreeMap` objects instead of `Hashmap`, you wouldn't need to sort the keyset you retrieve here. However I'm not sure if the cost of lookups/inserts going from O(1) to O(logN) would be worth that trade off.